src/streamsync/core_ui.py

```diff
@@ -157,11 +182,41 @@ def to_dict(self) -> Dict:
             for component_id, base_component
             in self.base_component_tree.components.items()
         }
-        for component_id, session_component in self.components.items():
-            # Overriding base tree components with session-specific ones
-            active_components[component_id] = session_component.to_dict()
+        for component_id, own_component in self.components.items():
+            # Overriding base tree components with ones that belong to dependent tree
+            active_components[component_id] = own_component.to_dict()
         return active_components
 
+
+class SessionComponentTree(DependentComponentTree):
+
+    def __init__(self, base_component_tree: ComponentTree, base_cmc_tree: ComponentTree):
+        super().__init__(base_component_tree, attach_root=False)
+
+        # Initialize SessionComponentTree with components from the base
+        # CMC pool, added during app initialization
+        preinitialized_components = base_cmc_tree.to_dict()
+        self.ingest(preinitialized_components)
+        self.counter = len(self.components)
```

Comment:
What are we using the counter for? I remember I set it up initially, but I'm not sure if it's used

Reply:
To be honest, I haven't seen it being used, but kept it just to be safe in case I missed it somewhere.

Reply:
Cool, can you please do a `.counter` search and see how that goes? That should cover it

Reply:
<img width="281" alt="image" src="https://github.com/streamsync-cloud/streamsync/assets/157150795/a7bb1a9f-f321-4f03-adba-f5322d6d2ecf">
Seems to be confined to these classes indeed, safe to remove.

Reply:
Nice, let's remove